docs/references/architecture/adr-112-txi.md

```diff
@@ -0,0 +1,84 @@
+# ADR 112: Modular transaction hashing
+
+## Changelog
+
+- 2024-02-05: First version (@melekes)
+
+## Status
+
+TBD
+
+## Context
+
+Transaction hashing in CometBFT is currently implemented using `tmhash`
+package, which itself relies on `sha256` to calculate a transaction's hash.
+
+The hash is then used by the built-in indexer (to index this
```

Comment:
Can you elaborate more this list?

The mempool implementation uses transaction hashes. Both to produce log messages and as a key to maps of transactions (`types.Tx.Key()` is a hash as an array with fixed size).